rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java

```diff
@@ -544,17 +554,63 @@ public ResponseEntity<TaskRepresentationModel> terminateTask(@PathVariable Strin
   public ResponseEntity<TaskRepresentationModel> transferTask(
       @PathVariable String taskId,
       @PathVariable String workbasketId,
-      @RequestBody(required = false) Boolean setTransferFlag)
+      @RequestBody(required = false)
+          TransferTaskRepresentationModel transferTaskRepresentationModel)
       throws TaskNotFoundException,
           WorkbasketNotFoundException,
           NotAuthorizedOnWorkbasketException,
           InvalidTaskStateException {
-    Task updatedTask =
-        taskService.transfer(taskId, workbasketId, setTransferFlag == null || setTransferFlag);
+    Task updatedTask;
+    if (transferTaskRepresentationModel == null) {
+      updatedTask = taskService.transfer(taskId, workbasketId);
+    } else {
+      updatedTask =
+          taskService.transferWithOwner(
+              taskId,
+              workbasketId,
+              transferTaskRepresentationModel.getOwner(),
+              transferTaskRepresentationModel.getSetTransferFlag());
+    }
 
     return ResponseEntity.ok(taskRepresentationModelAssembler.toModel(updatedTask));
   }
 
+  /**
+   * This endpoint transfers a list of Tasks listed in the body to a given Workbasket, if possible.
+   * Tasks that can be transfered without throwing an exception get transferred independent of other
+   * Tasks. If the transfer of a Task throws an exception, then the Task will remain in the old
+   * Workbasket.
+   *
+   * @title Transfer Tasks to another Workbasket
+   * @param workbasketId the Id of the destination Workbasket
+   * @param transferTaskRepresentationModel JSON formatted request body containing the TaskIds,
+   *     owner and setTransferFlag of tasks to be transferred; owner and setTransferFlag are
+   *     optional, while the TaskIds are mandatory
+   * @return the successfully transferred Task.
```

Comment:
The return comment is wrong. Please fix it!

Reply:
Yes, I will fix it